quadratic/quadratic_problem.py

```diff
@@ -65,6 +67,87 @@ def from_params(cls, G: np.ndarray, c: np.ndarray, A: np.ndarray,
                    solution=solution, x0=x0)
 
     @property
-    def is_inequality_constrained(self):
+    def is_inequality_constrained(self) -> bool:
         """Return whether or not any of the imposed constraints are inequalities."""
-        return any(constraint.equation_type is not EquationType.EQ for constraint in self.constraints)
+        return any(constraint.equation_type is not EquationType.EQ
+                   for constraint in self.constraints)
+
+    def find_x0(self, initial_guess: np.ndarray = None) -> np.ndarray:
```

Comment:
btw - imo hat minimisierungslogik nix in an problem verlorn, da diese indep. von da minimisierung selbst sein sollten. z.b., viell. möchte man SQP verwendn für ein QuadraticProblem, dann hätte da find_x0 auf basis von QP nix zu suchn

Reply:
To be fair, i persönlich find schon, dass den Startpunkt finden was mit dem Problem zu tun hat, weil es bound zu de constraints is. I hätt persönlich wahrscheinlich `find_x0` zu einer methode in der base klasse gmacht und glei irgendwo im konstruktor gerufen, damit des interface von allem klein bleibt

Reply:
Kann ma dann schön inheriten und gegebenfalls a von der superklasse überschreiben oder aufrufen; als eigene funktion hats mmN kan Mehrwert, weil is außerhalb der Probleme ned benutz. Aber wir kennans a in a funktion schreiben und einfach constraints übergeben

Reply:
Ja eh, aber es is mehr so a Datenklasse^^ Da sollt ja eig 0 Logik drin sein, was wir eh schon bissl violaten, durch diverse calc-Methoden etc. Da kann man sich vermutlich auch drüber streiten, was gehört in ne Datenklasse und was nicht. Theoretisch könnt ma ja die ganze Active-Set-QP in QuadraticProblem schreiben, wir haben uns aber dazu entschieden, Minimisierungslogik in separaten Python-Files abzuhandeln. Dann wäre es imo sauberer, diesen Teil der Minimisierung auch separat zu handlen. 

Aber sonst könn ma's schon drinlassn

Reply:
I hab einfach überlegt, wie i die Methode/Funktion nennen soll, und es wär was in Richtung `find_x0_qp` worden, was einfach schlechten code smell hat iMo, weils halt eh mit QPs was zu tun hat, und so hat ma quasi no a extra herumschwebende funktion. I glaub es is schon bissl overhead, wenn i aus am anderen submodule was importieren muss, was im Konstruktor automatisch gehandelt werden kann